File: src/com/bradhouse/differentiate/Nodes/TreeNode.java
package com.bradhouse.differentiate.Nodes;

/**
 * Created by brad on 5/22/16.
 */
public abstract class TreeNode {
    TreeNode left;
    TreeNode right;

    public TreeNode getLeft() {
        return left;
    }

    public TreeNode getRight() {
        return right;
    }

    public boolean isLeaf() {
        return false;
    }

    public boolean isSame(TreeNode n) {
        return this.getClass() == n.getClass() && left == n.left && right == n.right;
    }

//    abstract TreeNode diff();
}

File: src/main/java/com/bradhouse/differentiate/Nodes/TreeNode.kt
package com.bradhouse.differentiate.Nodes

abstract class TreeNode {
    var left: TreeNode? = null
    var right: TreeNode? = null

    open val isLeaf = false

    fun isSame(n: TreeNode): Boolean {
        return this.javaClass == n.javaClass && left == n.left && right == n.right
    }

    override fun toString(): String {
        return ""
    }

}
